internal/executor/shell_unix_test.go

```diff
@@ -8,6 +8,10 @@ import (
 	"time"
 )
 
+func TestZshDoesNotHang(t *testing.T) {
+	_, _ = ShellCommandsAndGetOutput([]string{"zsh -c 'echo \"a:b\" | read -d \":\" piece'"}, nil, nil)
```

Comment:
Let's either add `timeout` here or to the Cirrus task so in case of hanging it won't hang for an hour

Reply:
Added a timeout in https://github.com/cirruslabs/cirrus-ci-agent/pull/98/commits/d41b9382b190b7dc7f4368d7710cee626ba8b2e0.